DSharpPlus/Net/Rest/DiscordApiClient.cs

```diff
@@ -1044,7 +1044,7 @@ internal async Task<DiscordMessage> GetChannelMessageAsync(ulong channel_id, ulo
             return ret;
         }
 
-        internal async Task<DiscordMessage> EditMessageAsync(ulong channel_id, ulong message_id, Optional<string> content, Optional<IEnumerable<DiscordEmbed>> embeds, IEnumerable<IMention> mentions, IReadOnlyList<DiscordActionRowComponent> components)
+        internal async Task<DiscordMessage> EditMessageAsync(ulong channel_id, ulong message_id, Optional<string> content, Optional<IEnumerable<DiscordEmbed>> embeds, IEnumerable<IMention> mentions, IReadOnlyList<DiscordActionRowComponent> components, IReadOnlyCollection<DiscordMessageFile> files)
```

Comment:
Why not IEnumerable

Reply:
`DoMultipartAsync` takes `IReadOnlyCollection`